src/frontend/src/eth/utils/qr-code.utils.ts

```diff
@@ -19,13 +23,58 @@ export const decodeQrCode = ({
 		return { status: 'cancelled' };
 	}
 
-	const payment = decodePayment(code);
+	const payment = decodeQrCodeUrn(code);
 
 	if (isNullish(payment)) {
 		return { status: 'success', destination: code };
 	}
 
-	const { token, identifier: destination, amount } = payment;
+	const { prefix, destination, amount, address, ethereumChainId, functionName } = payment;
+
+	const normalizeChainId = (chainId: string): string => {
+		if (chainId.startsWith('0x')) {
+			return parseInt(chainId, 16).toString();
+		}
+		return chainId;
+	};
+
+	const matchEthereumToken = (
```

Comment:
can we use an object as parameter for consistency reason?

Reply:
I didn't quite understand the question. You mean to pass `matchEthereumToken` as property?

Reply:
```
const matchEthereumToken = (address: string | undefined,
		functionName: string | undefined,
		ethereumChainId: string | undefined
	): string | undefined => {
```

=> (for example)

```
const matchEthereumToken = ({address, functionName, ethereumChainId}: {address: string | undefined,
		functionName: string | undefined,
		ethereumChainId: string | undefined}
	): string | undefined => {
```

Reply:
Ah! Right! I forgot sorry!